app/models/news.rb

```diff
@@ -30,9 +32,15 @@ def mark_read(user)
   def self.num_unread_for(user)
     return 0 unless user
 
-    view = NewsView.find_by(user_id: user.id)
-    return News.count unless view
+    Rails.cache.fetch(NewsView.cache_string_for(user.id), expires_in: 1.month) do
+      view = NewsView.find_by(user: user)
+      view ? News.where('id > ?', view.news_id).count : News.count
+    end
+  end
+
+  private
 
-    News.where('id > ?', view.news_id).count
+  def invalidate_caches
+    Rails.cache.delete_matched("#{Rails.env}.#{NewsView::CACHE_VERSION}.unread_news_count.*")
```

Comment:
Could we keep this on `NewsView` as a class method that we call from here? That way we can keep the patterns for each cache string centralized on a single model where possible (making it easier to spot accidental divergences).

I think a `NewsView.invalidate_all_caches` would be good!